File: readingeggs/app/workers/event_consumer_worker.rb
class EventConsumerWorker
  include Sidekiq::Worker

  def perform(base_event)
    # puts "=============="
    # puts "| Base Event |"
    # puts "=============="
    # puts base_event.inspect
    # puts "\n"
    # puts "==================="
    # puts "| Hydrating Event |"
    # puts "==================="
    # hydrated_event = hydrate(base_event)
    # puts hydrated_event.inspect
    # puts "\n"
    # puts "================"
    # puts "| Saving Event |"
    # puts "================"
    # puts "👌"
    # puts "\n"
  end

  private

  def hydrate(base_event)
    base_event.merge(add_facts)
  end

  def add_facts
    { "Garretts" => "awesome!" }
  end
end

File: student_event_processor/lib/student_event_processor/workers/event_consumer_worker.ex
defmodule EventConsumerWorker do
  def perform(base_event) do
    IO.puts "=============="
    IO.puts "| Base Event |"
    IO.puts "=============="
    IO.inspect base_event
    IO.puts "\n"
    IO.puts "==================="
    IO.puts "| Hydrating Event |"
    IO.puts "==================="
    hydrated_event = hydrate(base_event)
    IO.inspect hydrated_event
    IO.puts "\n"
    IO.puts "================"
    IO.puts "| Saving Event |"
    IO.puts "================"
    IO.puts "👌"
    IO.puts "\n"
  end

  defp hydrate(base_event) do
    base_event
    |> Map.merge(add_facts())
  end

  defp add_facts, do: %{"Garretts" => "awesome!"}
end
